File: 118-pascals-triangle/118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        
        pascal = []
        
        for i in range(0, numRows):
            ans = []
            for j in range(0, i+1):
                if j == 0 or j == i:
                    ans.append(1)
                else:
                    val = pascal[i-1][j-1] + pascal[i-1][j]
                    ans.append(val)
            
            pascal.append(ans)
            
        return pascal
    

File: 118-pascals-triangle/118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        
        vector<vector<int>> ans;
        
        for(int i=0; i<numRows; i++){
            
            vector<int> col;
            
            for(int j=0; j<=i; j++){
                
                if(j == 0 || j == i){
                    col.push_back(1);
                }
                else{
                    int val = ans[i-1][j-1] + ans[i-1][j];
                    col.push_back(val);
                }
                
            }
            
            ans.push_back(col);
        }
        
        return ans;
    }
};